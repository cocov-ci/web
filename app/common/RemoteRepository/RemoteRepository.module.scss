@import "styles/variables.module";

.base {
  padding: 14px 30px;
  display: flex;
  border-bottom: 1px solid $colorWhiteDark;

  .info {
    margin-left: 16px;
  }

  .icon {
    margin-top: 8px;
  }

  .name {
    font-variation-settings: "wght" 650;
    font-size: 24px;
    color: $colorBlack;
    letter-spacing: 0;
    text-align: left;
  }

  .description.empty {
    opacity: 0.45;
  }

  .dates {
    opacity: 0.45;
  }

  .button {
    flex: 1;
    display: flex;
    justify-content: flex-end;
    align-content: flex-start;
    flex-wrap: wrap;
  }
}
