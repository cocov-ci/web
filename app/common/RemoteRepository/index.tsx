import classNames from 'classnames'
import { Album } from 'lucide-react'
import React from 'react'

import Button from '../Button'
import RelativeTime from '../RelativeTime'

import styles from './RemoteRepository.module.scss'

type RemoteRepositoryProps = {
  children?: React.ReactNode
  className?: string
  name: string
  description?: string
  createdAt: Date
  updatedAt?: Date
  status: 'absent' | 'pending' | 'present'
  onAdd?: () => void
}

const RemoteRepository = ({
  className,
  name,
  description,
  createdAt,
  updatedAt,
  status,
  onAdd,
}: RemoteRepositoryProps) => {
  let addButton

  if (status == 'absent') {
    addButton = (
      <Button onClick={onAdd} style="secondary">
        Add to Cocov
      </Button>
    )
  } else if (status == 'pending') {
    addButton = (
      <Button disabled={true} style="secondary">
        Adding...
      </Button>
    )
  } else {
    addButton = (
      <Button disabled={true} style="inactive">
        Already Added
      </Button>
    )
  }

  return (
    <div className={classNames(styles.base, className)}>
      <div className={styles.icon}>
        <Album size={25} strokeWidth="2px" />
      </div>
      <div className={styles.info}>
        <div className={styles.name}>{name}</div>
        <div
          className={classNames(styles.description, {
            [styles.empty]: !description,
          })}
        >
          {description || 'No description'}
        </div>
        <div className={styles.dates}>
          Created <RelativeTime timestamp={createdAt} />
          {updatedAt && (
            <>
              {' '}
              â€¢ Last updated <RelativeTime timestamp={updatedAt} />
            </>
          )}
        </div>
      </div>
      <div className={styles.button}>{addButton}</div>
    </div>
  )
}

export default RemoteRepository
