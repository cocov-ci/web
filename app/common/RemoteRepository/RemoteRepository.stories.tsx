import type { Meta, StoryObj } from '@storybook/react'

import RemoteRepository from './index'

const meta: Meta<typeof RemoteRepository> = {
  title: 'Common/RemoteRepository',
  component: RemoteRepository,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof RemoteRepository>

export const Default: Story = {
  args: {
    name: 'api',
    createdAt: new Date('2023-02-20T20:21:22Z'),
    updatedAt: new Date('2023-02-20T21:21:22Z'),
    status: 'absent',
  },
}

export const Pending: Story = {
  args: {
    name: 'api',
    createdAt: new Date('2023-02-20T20:21:22Z'),
    updatedAt: new Date('2023-02-20T21:21:22Z'),
    status: 'pending',
  },
}

export const Present: Story = {
  args: {
    name: 'api',
    createdAt: new Date('2023-02-20T20:21:22Z'),
    updatedAt: new Date('2023-02-20T21:21:22Z'),
    status: 'present',
  },
}
