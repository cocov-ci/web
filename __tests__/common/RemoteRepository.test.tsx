import { render, screen } from '@testing-library/react'

import RemoteRepository from 'app/common/RemoteRepository'

import '@testing-library/jest-dom'

const args = {
  name: 'api',
  createdAt: new Date(new Date().setDate(new Date().getDate() - 2)),
  updatedAt: new Date(new Date().setDate(new Date().getDate() - 1)),
  status: 'absent' as const,
}

describe('common/RemoteRepository', () => {
  it('renders RemoteRepository', () => {
    render(<RemoteRepository {...args} />)

    expect(screen.getByText('api')).toBeVisible()
    expect(screen.getByText('yesterday')).toBeVisible()
    expect(screen.getByText('2 days ago')).toBeVisible()
    expect(screen.getByRole('button')).toHaveTextContent('Add to Cocov')
  })

  it('renders RemoteRepository', () => {
    render(<RemoteRepository {...args} status="pending" />)

    expect(screen.getByRole('button')).toHaveTextContent('Adding...')
  })

  it('renders RemoteRepository', () => {
    render(<RemoteRepository {...args} status="present" />)

    expect(screen.getByRole('button')).toHaveTextContent('Already Added')
  })

  it('renders RemoteRepository snapshots', () => {
    const { container } = render(<RemoteRepository {...args} />)

    expect(container).toMatchSnapshot()
  })
})
