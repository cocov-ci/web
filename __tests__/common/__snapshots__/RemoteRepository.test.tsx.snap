// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/RemoteRepository renders RemoteRepository snapshots 1`] = `
<div>
  <div
    class="base"
  >
    <div
      class="icon"
    >
      <svg
        class="lucide lucide-album"
        fill="none"
        height="25"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2px"
        viewBox="0 0 24 24"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="18"
          rx="2"
          ry="2"
          width="18"
          x="3"
          y="3"
        />
        <polyline
          points="11 3 11 11 14 8 17 11 17 3"
        />
      </svg>
    </div>
    <div
      class="info"
    >
      <div
        class="name"
      >
        api
      </div>
      <div
        class="description empty"
      >
        No description
      </div>
      <div
        class="dates"
      >
        Created 
        <span
          class=""
          title="Mar 28, 2023, 11:59 PM GMT+1"
        >
          2 days ago
        </span>
         
        â€¢ Last updated 
        <span
          class=""
          title="Mar 29, 2023, 11:59 PM GMT+1"
        >
          yesterday
        </span>
      </div>
    </div>
    <div
      class="button"
    >
      <button
        class="className button secondary"
      >
        Add to Cocov
      </button>
    </div>
  </div>
</div>
`;
